### DISCUSSION TITLE
Select Component with Icons.

### DISCUSSION THREAD

**Discussion by Slddev** [saadeghi/daisyui#2976] [Ideas / feature requests]
![image](https://github.com/saadeghi/daisyui/assets/74999267/52243a4e-bc56-4952-83df-8b45b32a9c21)
Something like that but with select elements.

**Comment by saadeghi** [comment]
Styling `<select>` has a lot of limits unfortunately.  
We can't put an icon inside it
Also, unlike a text input where it gets focused when the parent label is focused, a `<select>` inside a `<label>` won't open up. It just gets focused. So we can't simply put an icon and a select inside a label and expect it to work similar to a text input.

But we can use some tricks. We can make the select absolute so it fills the whole area of label, then we can apply outline to the label when the child select is focused:  
https://play.tailwindcss.com/403RBcF6gs

Hopefully in the future we can have more customizing ability for `<select>` element when browsers start supporting it:  
https://open-ui.org/components/selectlist/
